tests/scripts/thread-cert/border_router/MATN_09_DefaultBRMulticastForwarding.py

```diff
@@ -0,0 +1,186 @@
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import logging
+import unittest
+
+import pktverify
+from pktverify import packet_verifier, packet_filter, consts
+from pktverify.consts import MA1
+import config
+import thread_cert
+
+# Test description:
+# The purpose of this test case is to verify that a Secondary BBR can take over
+# forwarding of outbound multicast transmissions from the Thread Network when
+# the Primary BBR disconnects. The Secondary in that case becomes Primary.
+#
+# Topology:
+#    ----------------(eth)------------------
+#           |                  |
+#         BR_1 (Leader) ---- BR_2
+#           |                  |
+#           |                  |
+#          ROUTER_1 -----------+
+#
+
+BR_1 = 1
+BR_2 = 2
+ROUTER_1 = 3
+
+
+class MATN_09_FailureOfPrimaryBBROutboundMulticast(thread_cert.TestCase):
+    USE_MESSAGE_FACTORY = False
+
+    TOPOLOGY = {
+        BR_1: {
+            'name': 'BR_1',
+            'is_otbr': True,
+            'allowlist': [BR_2, ROUTER_1],
+            'version': '1.2',
+            'router_selection_jitter': 2,
+        },
+        BR_2: {
+            'name': 'BR_2',
+            'is_otbr': True,
+            'allowlist': [BR_1, ROUTER_1],
+            'version': '1.2',
+            'router_selection_jitter': 2,
+        },
+        ROUTER_1: {
+            'name': 'Router_1',
+            'allowlist': [BR_1, BR_2],
+            'version': '1.2',
+            'router_selection_jitter': 2,
+            'partition_id': 1,
+        },
+    }
+
+    def test(self):
+        br1 = self.nodes[BR_1]
+        br2 = self.nodes[BR_2]
+        router1 = self.nodes[ROUTER_1]
+
+        br1.start()
+        self.simulator.go(5)
+        self.assertEqual('leader', br1.get_state())
+        self.assertTrue(br1.is_primary_backbone_router)
+
+        router1.start()
+        self.simulator.go(5)
+        self.assertEqual('router', router1.get_state())
+
+        br2.start()
+        self.simulator.go(5)
+        self.assertEqual('router', br2.get_state())
+        self.assertFalse(br2.is_primary_backbone_router)
+
+        # 1. Router sends a ping packet to the multicast address, MA1,
+        # encapsulated in an MPL packet
+        self.assertFalse(router1.ping(MA1))
+        self.simulator.go(5)
+
+        # 4a. Switch off BR_1
+        br1.disable_backbone_router()
+        br1.thread_stop()
+        br1.interface_down()
+        self.simulator.go(180)
+
+        # 4b. BR_2 Detects the missing Primary BBR and becomes the the Leader of
+        # the Thread Network.
+        self.assertEqual('disabled', br1.get_state())
+        self.assertEqual('leader', br2.get_state())
```

Comment:
I wonder why BR_2 must become leader in this case (even though the Test Spec. clearly states so). 
Can Router_1 become Leader even with lowest Partition ID? @librasungirl

Reply:
Configure a larger NETWORK_ID_TIMEOUT value (maybe 160s, larger than default 120s) for ROUTER_1, thus it takes longer for ROUTER_1 to detect the lost of Leader, and when it tries to reattach after NETWORK_ID_TIMEOUT, BR_2 has already formed a new partition.

Reply:
I am not sure if TestHarness did the same `NETWORK_ID_TIMEOUT` configuration. 
BTW, if we configure `NETWORK_ID_TIMEOUT`, there is no need to configure lowest partition ID on `Router_1`.

Reply:
> I am not sure if TestHarness did the same NETWORK_ID_TIMEOUT configuration.

May check test harness script for confirmation.

> BTW, if we configure NETWORK_ID_TIMEOUT, there is no need to configure lowest partition ID on Router_1.

Guess such joint configurations are to ensure the expecting topology - though Leader periodically send MLE advertisement with new router id sequence to link local multicast address, however whether or not neighboring devices hear it depends on the environment in real on-device test, that is, there is some uncertainty which device assumes the lost of the Leader first. Though the communication is more stable for simulation scenarios, there should be no harm to keep it.